lua/guard-collection/linter/hlint.lua

```diff
@@ -0,0 +1,35 @@
+local lint = require('guard.lint')
+
+local severities = {
+  suggestion = lint.severities.info,
+  warning = lint.severities.warning,
+  error = lint.severities.error,
+}
+
+return {
+  cmd = 'hlint',
+  args = { '--json', '--no-exit-code' },
+  fname = true,
+  parse = function(result, bufnr)
+    local diags = {}
+
+    result = result ~= '' and vim.json.decode(result) or {}
+    if type(result) == 'table' then
```

Comment:
This check is redundant because it is either valid json or {} by the previous line

Reply:
ah yeah, I should've used `vim.NIL` or maybe decoded the result without a check since it already returns a NIL if it fails. Thanks for fixing that